spark_client/services.py

```diff
@@ -112,6 +96,482 @@ def user(self):
+                    labels=labels_to_pass,
+                ),
+                buffer,
+            )
+            buffer.seek(0)
+            return list(yaml.safe_load_all(buffer))
+
+    def get_secret(self, secret_name: str, namespace: str) -> Dict[str, Any]:
+        """Return the data contained in the specified secret.
+
+        Args:
+            secret_name: name of the secret
+            namespace: namespace where the secret is contained
+        """
+
+        with io.StringIO() as buffer:
+            codecs.dump_all_yaml(
+                [self.client.get(res=Secret, namespace=namespace, name=secret_name)],
+                buffer,
+            )
+            buffer.seek(0)
+            secret = yaml.safe_load(buffer)
+
+            result = dict()
+            for k, v in secret["data"].items():
+                result[k] = base64.b64decode(v).decode("utf-8")
+
+            secret["data"] = result
+            return secret
+
+    def set_label(
+        self, resource_type: str, resource_name: str, label: str, namespace: str
+    ):
+        """Set label to a specified resource (type and name).
+
+        Args:
+            resource_type: type of the resource to be labeled, e.g. service account, rolebindings, etc.
+            resource_name: name of the resource to be labeled
+            namespace: namespace where the resource is
+        """
+
+        label_fragments = label.split("=")
+        patch = {"metadata": {"labels": {label_fragments[0]: label_fragments[1]}}}
+
+        if resource_type == "serviceaccount":
+            self.client.patch(
+                res=LightKubeServiceAccount,
+                name=resource_name,
+                namespace=namespace,
+                obj=patch,
+            )
+        elif resource_type == "role":
+            self.client.patch(
+                res=Role, name=resource_name, namespace=namespace, obj=patch
+            )
+        elif resource_type == "rolebinding":
+            self.client.patch(
+                res=RoleBinding, name=resource_name, namespace=namespace, obj=patch
+            )
+        else:
+            raise NotImplementedError(
+                f"Label setting for resource name {resource_type} not supported yet."
+            )
+
+    def remove_label(
+        self, resource_type: str, resource_name: str, label: str, namespace: str
+    ):
+        """Remove label to a specified resource (type and name).
+
+        Args:
+            resource_type: type of the resource to be labeled, e.g. service account, rolebindings, etc.
+            resource_name: name of the resource to be labeled
+            label: label to remove
+            namespace: namespace where the resource is
+        """
+        label_to_remove = f"/metadata/labels/{label.replace('/','~1')}"
+        self.logger.debug(f"Removing label {label_to_remove}")
+        patch = [{"op": "remove", "path": label_to_remove}]
+
+        if resource_type == "serviceaccount":
```

Comment:
**suggestion** We could use Enum for handling resource_type, given they are values chosen from a set of values

https://docs.python.org/3/library/enum.html

This should make it more typed, and also avoid to use strings, which are error prone